dask_expr/expr.py

```diff
@@ -494,6 +504,8 @@ def __dask_graph__(self):
             for operand in expr.operands:
                 if isinstance(operand, Expr):
                     stack.append(operand)
+                elif not isinstance(expr, Fused) and isinstance(operand, list):
```

Comment:
I think the check for ``list`` is too generic (same above). We could convert a list of Expr objects into a custom object that we could check for here (just as one example).

Is there already a solution for handling a list of ``Expr`` objects for one argument? ``concat([df, df2])`` or ``join`` for example